server/games/typedefs.py

```diff
@@ -0,0 +1,177 @@
+from enum import Enum, unique
+from typing import Dict, List, NamedTuple, Optional, Set
+
+from server.games.game_results import GameOutcome
+from server.players import Player
+from server.rating import RatingType
+
+
+@unique
+class GameState(Enum):
+    INITIALIZING = 0
+    LOBBY = 1
+    LIVE = 2
+    ENDED = 3
+
+
+@unique
+class Victory(Enum):
+    DEMORALIZATION = 0
+    DOMINATION = 1
+    ERADICATION = 2
+    SANDBOX = 3
+
+    @staticmethod
+    def from_gpgnet_string(value: str) -> Optional["Victory"]:
```

Comment:
I learned recently that this can be done more easily with `Victory[value.upper()]` (maybe need to catch `KeyError` depending on what the caller bahavior is)